src/server-extension/resolvers/AdminResolver/index.ts

```diff
@@ -24,36 +50,45 @@ import {
   VideoViewPerUserTimeLimit,
   VideoWeights,
 } from './types'
-import { config, ConfigVariable } from '../../../utils/config'
-import { OperatorOnly } from '../middleware'
-import {
-  Video,
-  VideoCategory,
-  VideoFeaturedInCategory,
-  VideoHero as VideoHeroEntity,
-} from '../../../model'
-import { GraphQLResolveInfo } from 'graphql'
-import { Context } from '@subsquid/openreader/lib/context'
-import { parseObjectTree } from '@subsquid/openreader/lib/opencrud/tree'
-import { getResolveTree } from '@subsquid/openreader/lib/util/resolve-tree'
-import { EntityByIdQuery } from '@subsquid/openreader/lib/sql/query'
-import { getObjectSize } from '@subsquid/openreader/lib/limit.size'
-import { VideoHero } from '../baseTypes'
-import { model } from '../model'
-import { ed25519PairFromString, ed25519Sign } from '@polkadot/util-crypto'
-import { u8aToHex, hexToU8a, isHex } from '@polkadot/util'
-import { generateAppActionCommitment } from '@joystream/js/utils'
-import { AppAction } from '@joystream/metadata-protobuf'
-import { withHiddenEntities } from '../../../utils/sql'
 import { processCommentsCensorshipStatusUpdate } from './utils'
-import { videoRelevanceManager } from '../../../mappings/utils'
 
 @Resolver()
 export class AdminResolver {
-  // Set by depenency injection
+  // Set by dependency injection
   constructor(private em: () => Promise<EntityManager>) {}
 
-  @UseMiddleware(OperatorOnly)
+  @UseMiddleware(OperatorOnly(OperatorPermission.GRANT_OPERATOR_PERMISSIONS))
+  @Mutation(() => Boolean)
+  async grantPermissions(@Args() args: GrantOperatorPermissionsInput): Promise<boolean> {
+    const em = await this.em()
+    const user = await em.findOne(User, { where: { id: args.userId } })
+    if (!user) {
+      throw new Error('User not found')
+    }
+
+    // Add only new permissions that the user doesn't have yet
+    user.permissions = Array.from(new Set([...(user.permissions || []), ...args.permissions]))
+
+    await em.save(user)
+    return true
+  }
+
+  @UseMiddleware(OperatorOnly(OperatorPermission.REVOKE_OPERATOR_PERMISSIONS))
+  @Mutation(() => Boolean)
```

Comment:
Maybe return the new user permission set?
From a user perspective I guess the raw boolean is a bit cryptic, I guess, but I can be wrong, plus all the `set*` mutation that we already have returns the new value just set eg `{isKilled: bool}` for the `setKillswitch` mutation